lib/src/decimal128.dart

```diff
@@ -0,0 +1,149 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:convert';
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+
+import 'native/realm_bindings.dart';
+import 'native/realm_core.dart';
+
+/// A 128-bit decimal floating point number.
+class Decimal128 extends Comparable<Decimal128> {
+  /// The value 0.
+  static final zero = Decimal128.fromInt(0);
+
+  /// The value 1.
+  static final one = Decimal128.fromInt(1);
+
+  /// The value 10.
+  static final ten = Decimal128.fromInt(10);
```

Comment:
Wondering, why 1 and 10 are special and have them as properties.

Reply:
I expect 0,1 and 10 to be the most used constants, so I const construct them. Obviously the user is free not to use them, or define const constructed values themself. Problem is the Decimal.fromInt cannot be a const constructor, so it is more memory efficient to use `Decimal128.one` vs `Decimal.fromInt(1)` repeatedly.. and also ever so slightly faster

Reply:
Why do we expect these to be frequently used?

Reply:
Being the additive identity, multiplicative identity, and the base (0, 1, and 10). It also matches https://pub.dev/documentation/decimal/latest/decimal/Decimal-class.html